src/ExpressionEvaluator/Core/Source/ExpressionCompiler/PseudoVariableUtilities.cs

```diff
@@ -54,5 +54,14 @@ internal static DkmClrCompilationResultFlags GetLocalResultFlags(this Alias alia
                     return DkmClrCompilationResultFlags.None;
             }
         }
+
+        internal static bool IsReturnValue0(this Alias alias)
```

Comment:
Actually you want to filter out "$ReturnValue".  $ReturnValue is an alias for the last return value (largest number)